### PR TITLE
Cloud branch

### PR DESCRIPTION
**Pull request by alexreigle**
Discussion is needed to reconcile the merge conflicts. They involve both TNT and Clouds related code. Let's review the pull request and both agree on the changes before either of us decide on the final changes.

### CODE REVIEW

File: src/main.cpp
Comment (mbozada) [inline src/main.cpp:2255]: So this seems out of date from my last TNT related PR. I think that's where the merge conflict comes from. You can take a crack at trying to fix it, or we can hold off until I pull TNT related functions into separate source/header files. Up to you.

File: src/main.cpp
Comment (mbozada) [inline src/main.cpp:2283]: Why did you swap the function placement? I don't mind, but I want to know if I was making a mistake with how I had them before.

File: src/map.cpp
Comment (mbozada) [inline src/map.cpp]: Just wanted to point out that there's a lot of commented out code here. Also, this print statement is kind of uninformative.

We need to figure out how we want to do testing. I'm okay with print statement testing, but we'll probably have to come up with a format.

File: src/util.c
Comment (mbozada) [inline src/util.c:39]: Going from C to C++ do we need to typecast the alloc's? I'll probably not mess around with it based on what I'm doing, but I wouldn't mind learning the "why".

File: src/clouds.h
Comment (mbozada) [inline src/clouds.h]: I think these can be taken out. In fact, it may be good to go through all of the header/source files you've changed to double check that you're not over-including stuff. Fogleman seems to put the include statements in the source files, with no mention of them in header files.

**Review by mbozada — COMMENTED** [review]
I was able to compile and run it on my end, and it seems to be working well!

I have some questions and concerns that I've peppered throughout. The main things being that I think you need to double check your include statements and where you've commented out code. More details in other comments.

Also, we need to address the merge conflict. It seems that this doesn't have the recursive call for TNT's chain explosions. I can fix it or you can, I'll leave it up to you to decide.

File: src/main.cpp
Comment (alexreigle) [reply src/main.cpp:2255]: I think you should update the code to reflect your most recent pull request to the dev branch. I'll make the same changes to my CluodBranch after the fact.

File: src/main.cpp
Comment (alexreigle) [reply src/main.cpp:2283]: I don't recall. It may have been a C++ function called before definition type of thing.

File: src/map.cpp
Comment (alexreigle) [reply src/map.cpp]: This is just an artifact of me trying to begin implementing the cloud functions. I decided that it would be best to comment it out for now - since the code was stable. I'll be working on the implementation in this same spot in the code in the future so I don't think there is a problem with leaving it here for now.

Comment (alexreigle) [reply src/map.cpp]: I am currently building tests into the cloud.h/cloud.cpp files to start off with. But we will need to settle on a testing method - this seems like a topic for a meeting though.

File: src/util.c
Comment (alexreigle) [reply src/util.c:39]: I think there are several places that I've had to do some casting to make functions C++ compatible. Mostly the functions were trying to return a Null pointer instead of a pointer to whatever we need (i.e. a char in this case).

**Comment by alexreigle** [comment]
I've addressed the comments by removing the commented out code.

File: src/main.cpp
Comment (mbozada) [reply src/main.cpp:2283]: It probably was. I went ahead and fixed it by adding a function declaration. Kind of sloppy of me, but since I'm planning to extract the TNT functions to their own files soon anyway.

**Review by mbozada — APPROVED** [review]
Looks good to me.

**Pull request merged**
